### ISSUE TITLE
Problem with Redis message queue, not able to deploy

### ISSUE THREAD

**Issue by fabulous-nils** [miguelgrinberg/Flask-SocketIO#907]
I have a lot of trouble integrating Flask-SocketIO into my existing Flask-application.
First of all I have a problem that seems to be identical to #618. As soon as I add a redis message queue to the SocketIO initialization, no messages are send to the client. 
The relevant code I have for setting up Flask-SocketIO:
```python
#Start of main file
import eventlet
eventlet.monkey_patch()

[...]

socketio = SocketIO()
socketio.init_app(app, 
                  message_queue="redis://",
                  path=config.BASE_ROUTE + "/data/socket",
                  logger=True,
                  engineio_logger=True,
                  async_mode="eventlet")

[...]

socketio.run(app, host="0.0.0.0", port=8087, debug=True)
```

The log states that messages are send to the client, no errors. As soon as I comment out the message_queue parameter, it works.

As opposed to #618, I have the same problem regardless of usage of the `debug` parameter, and regardless of the operating system. I tried it both with Linux and Windows without success. The same Flask application is using the Redis database on different parts with success, so the problem is not connected to the Redis database server.

I wanted to try out different setups to maybe find a workaround or track down the problem, but unfortunately neither of the setups did work:

- With gevent+uwsgi I get a `ValueError: fd must be non-negative: -1` on startup
- With gunicorn+eventlet I get alternating `OSError: [Errno 9] Bad file descriptor` and `OSError: [Errno 107] Transport endpoint is not connected` as soon as a websocket client connects and the following log output: 
```
Server initialized for eventlet.
Invalid session 2c73bc1e19974a3cb131afdf4bfeb2b7
Invalid session 2c73bc1e19974a3cb131afdf4bfeb2b7
2c73bc1e19974a3cb131afdf4bfeb2b7: Failed websocket upgrade, expected UPGRADE packet, received None instead.
[2019-02-21 12:38:27 +0000] [9] [ERROR] Socket error processing request.
```
- with `async_mode="threading"` it is working, but that is not really a solution

Right now I ran out of ideas and am a little stuck at what else I could try.

**Comment by miguelgrinberg** [comment]
Not sure what those file descriptor errors are about, but clearly something is wrong, since you get a somewhat similar error from gevent+uwsgi  and from gunicorn+eventlet, which are completely different stacks. Does the stack trace of those errors give any additional clues?

As far as the problem with Redis, the issue you referenced occurred only on Windows. Are you also on Windows?

**Comment by fabulous-nils** [comment]
I am developing on Windows and deploying in Docker Linux containers and I got the issue in both environments. I just spend the last hours trying to build a minimal working example that shows the problem and now the issue is gone and everythin works. Unfortunately, I have no idea what caused it and, more importantly, what solved it.

For the file descriptor errors it was indeed a rather convoluted problem with my overall setup that I was able to resolve. Thanks for your support!
